rp2040-hal/src/pwm/mod.rs

```diff
@@ -373,6 +373,147 @@ where
+        &mut self,
+        _: &Slice<S, M>,
+    ) {
+        match S::DYN {
+            DynSliceId { num } if num == 0 => unsafe {
+                (*pac::PWM::ptr()).inte.write(|w| w.ch0().clear_bit())
+            },
+            DynSliceId { num } if num == 1 => unsafe {
+                (*pac::PWM::ptr()).inte.write(|w| w.ch1().clear_bit())
+            },
+            DynSliceId { num } if num == 2 => unsafe {
+                (*pac::PWM::ptr()).inte.write(|w| w.ch2().clear_bit())
+            },
+            DynSliceId { num } if num == 3 => unsafe {
+                (*pac::PWM::ptr()).inte.write(|w| w.ch3().clear_bit())
+            },
+            DynSliceId { num } if num == 4 => unsafe {
+                (*pac::PWM::ptr()).inte.write(|w| w.ch4().clear_bit())
+            },
+            DynSliceId { num } if num == 5 => unsafe {
+                (*pac::PWM::ptr()).inte.write(|w| w.ch5().clear_bit())
+            },
+            DynSliceId { num } if num == 6 => unsafe {
+                (*pac::PWM::ptr()).inte.write(|w| w.ch6().clear_bit())
+            },
+            DynSliceId { num } if num == 7 => unsafe {
+                (*pac::PWM::ptr()).inte.write(|w| w.ch7().clear_bit())
+            },
+            _ => unreachable!(),
+        }
+    }
+
+    /// Did the indicated slice overflow?
+    pub fn is_overflow<S: SliceId, M: SliceMode + ValidSliceMode<S>>(
+        &self,
+        _: &Slice<S, M>,
+    ) -> bool {
+        match S::DYN {
+            DynSliceId { num } if num == 0 => unsafe {
+                (*pac::PWM::ptr()).ints.read().ch0().bit_is_set()
+            },
+            DynSliceId { num } if num == 1 => unsafe {
+                (*pac::PWM::ptr()).ints.read().ch1().bit_is_set()
+            },
+            DynSliceId { num } if num == 2 => unsafe {
+                (*pac::PWM::ptr()).ints.read().ch2().bit_is_set()
+            },
+            DynSliceId { num } if num == 3 => unsafe {
+                (*pac::PWM::ptr()).ints.read().ch3().bit_is_set()
+            },
+            DynSliceId { num } if num == 4 => unsafe {
+                (*pac::PWM::ptr()).ints.read().ch4().bit_is_set()
+            },
+            DynSliceId { num } if num == 5 => unsafe {
+                (*pac::PWM::ptr()).ints.read().ch5().bit_is_set()
+            },
+            DynSliceId { num } if num == 6 => unsafe {
+                (*pac::PWM::ptr()).ints.read().ch6().bit_is_set()
+            },
+            DynSliceId { num } if num == 7 => unsafe {
+                (*pac::PWM::ptr()).ints.read().ch7().bit_is_set()
+            },
+            _ => unreachable!(),
+        }
+    }
+
+    /// Clear the overflow bit associated with this slice.
+    pub fn clear_overflow<S: SliceId, M: SliceMode + ValidSliceMode<S>>(
+        &mut self,
+        _: &Slice<S, M>,
+    ) {
+        match S::DYN {
+            DynSliceId { num } if num == 0 => unsafe {
+                (*pac::PWM::ptr()).intr.write(|w| w.ch0().clear_bit())
```

Comment:
The `INTR` register has been marked with type `WC` meaning that in order to clear the interrupt bit, 1 should be written. I.e. this should be
```suggestion
                (*pac::PWM::ptr()).intr.write(|w| w.ch0().set_bit())
```
to write only the bit we want to clear.